src/app/portlets/dot-edit-page/content/services/dot-edit-content-html/dot-edit-content-html.service.ts

```diff
@@ -387,10 +387,38 @@ export class DotEditContentHtmlService {
         this.dotEditContentToolbarHtmlService.addContentletMarkup(doc);
     }
 
+    private setScriptProperties(script: HTMLScriptElement, node: any): HTMLScriptElement {
+        script.type = 'text/javascript';
+
+        if (node.src) {
+            script.src = node.src;
+        } else {
+            script.text = node.textContent;
+        }
+        return script;
+    }
+
+    private getScriptTags(scriptTags, contentDivWrapper: HTMLElement) {
+        const doc = this.getEditPageDocument();
+
+        Array.from(contentDivWrapper.children).forEach((node: any) => {
+            if (node.tagName === 'SCRIPT') {
+                let script = doc.createElement('script');
+                script = this.setScriptProperties(script, node);
+                scriptTags.push(script);
+            } else if (node.children.length) {
+                this.getScriptTags(scriptTags, node);
+            }
+        });
+
+        return scriptTags;
+    }
+
     private appendNewContentlets(contentletEl: any, html: string): void {
+
         const contentletContentEl = contentletEl.querySelector('.dotedit-contentlet__content');
         contentletContentEl.innerHTML = ''; // Removing the loading indicator
-
+        let scriptTags: HTMLScriptElement[] = [];
```

Comment:
you don't need init it with a empty array, because later you overwrite this value with

https://github.com/dotCMS/core-web/pull/714/files#diff-e2760b805d923abc4137bf2e3ddf0ee9R430

Reply:
it needs to be initialized, because we are passing it, and using `push` in here: https://github.com/dotCMS/core-web/pull/714/files#diff-e2760b805d923abc4137bf2e3ddf0ee9R408